services/motion/localizer.go

```diff
@@ -56,8 +61,62 @@ type movementSensorLocalizer struct {
 // NewMovementSensorLocalizer creates a Localizer from a MovementSensor.
 // An origin point must be specified and the localizer will return Poses relative to this point.
 // A calibration pose can also be specified, which will adjust the location after it is calculated relative to the origin.
-func NewMovementSensorLocalizer(ms movementsensor.MovementSensor, origin *geo.Point, calibration spatialmath.Pose) Localizer {
-	return &movementSensorLocalizer{MovementSensor: ms, origin: origin, calibration: calibration}
+func NewMovementSensorLocalizer(ms movementsensor.MovementSensor, origin *geo.Point, calibration spatialmath.Pose) (Localizer, error) {
+	if err := validateGeopoint(origin); err != nil {
+		return nil, err
+	}
+
+	if err := validatePose(calibration); err != nil {
+		return nil, err
+	}
+
+	return &movementSensorLocalizer{MovementSensor: ms, origin: origin, calibration: calibration}, nil
+}
+
+// validateGeopoint validates that a geopoint can be used for
+// motion planning.
+func validateGeopoint(gp *geo.Point) error {
+	if math.IsNaN(gp.Lat()) {
+		return errors.New("lat can't be NaN")
+	}
+
+	if math.IsNaN(gp.Lng()) {
+		return errors.New("lng can't be NaN")
+	}
+
+	return nil
+}
+
+// validatePose validates that a pose can be used for
+// motion planning.
+func validatePose(p spatialmath.Pose) error {
```

Comment:
This should never be possible we control the construction of Poses pretty tightly, so I'm not a fan of adding this extra overhead in.